shell/src/main/java/org/apache/accumulo/shell/format/DeleterFormatter.java

```diff
@@ -79,33 +79,36 @@ public String next() {
     Mutation m = new Mutation(key.getRow());
     String entryStr = formatEntry(next, isDoTimestamps());
     boolean delete = force;
-    try {
-      if (!force) {
-        shellState.getReader().flush();
+    if (!force) {
+      try {
+        shellState.getWriter().flush();
+        // this will cause end of file Exception for one of the formatter tests
         String line = shellState.getReader().readLine("Delete { " + entryStr + " } ? ");
         more = line != null;
         delete = line != null && (line.equalsIgnoreCase("y") || line.equalsIgnoreCase("yes"));
```

Comment:
Could the portion before `&&` be replaced with `more`?
```suggestion
        delete = more && (line.equalsIgnoreCase("y") || line.equalsIgnoreCase("yes"));
```

Reply:
I will add this. Sorry for the overlook.